trie/database.go

```diff
@@ -79,7 +82,6 @@ type Database struct {
 	newest  common.Hash                 // Newest tracked node, flush-list tail
 
 	preimages map[common.Hash][]byte // Preimages of nodes from the secure trie
-	seckeybuf [secureKeyLength]byte  // Ephemeral buffer for calculating preimage keys
```

Comment:
AFAIK this was a deliberate optimization to avoid reallocating it for every trie node change. It should still be usable in Cap and Commit since those are single threaded. Shouldn't we just make `preimage` not use it?

Reply:
In this PR, the Cap and Commit operations reuse a temporary buffer. so it's not "one per trie node change", but one per Cap / Commit.

Reply:
But, yeah, you're right, that would also make the change a lot smaller, so I'm inclined to agree with you there

Reply:
On second thought, I disagree a bit. `Cap`/`Commit` isn't recursive, right? So it's only every once in a while? 
Then I think it's saner to use the approach in my PR, instead of having an internal corruption-waiting-to-happen. Makes it a bit more robust against future changes. 
Thoughts?

Reply:
That's a fair point, didn't see that you reused a temporary one inside cap/commit.